tests/aws-cpp-sdk-cognitoidentityprovider-integration-tests/OperationTest.cpp

```diff
@@ -0,0 +1,379 @@
+        }
+        return {clientId,clientSecret};
+    }
+
+    bool deletePoolClient(const Aws::String& userPoolId, const Aws::String& clientId)
+    {
+        // Set up the request for deleting the User Pool Client
+        Aws::CognitoIdentityProvider::Model::DeleteUserPoolClientRequest request;
+        request.SetUserPoolId(userPoolId);
+        request.SetClientId(clientId);
+
+        // Make the call to delete the user pool client
+        auto outcome = client->DeleteUserPoolClient(request);
+        if (outcome.IsSuccess())
+        {
+            AWS_LOGSTREAM_DEBUG(ALLOCATION_TAG, "User Pool Client deleted successfully.");
+        }
+        return outcome.IsSuccess();
+    }
+
+};
+
+
+Aws::String ComputeSecretHash(const Aws::String &userPoolClientId, const Aws::String &clientSecret, const Aws::String &userName)
+{
+    const auto message = userName + userPoolClientId;
+    unsigned char hash[EVP_MAX_MD_SIZE];
+    unsigned int len = 0;
+
+    HMAC(EVP_sha256(),
+    clientSecret.c_str(), clientSecret.length(),
+    (unsigned char*)message.c_str(), message.length(),
+    hash, &len);
+    
+    // Base64 encode the hash
+    char encoded[128];
+    EVP_EncodeBlock((unsigned char*)encoded, hash, len);
+
+    return std::string(encoded);
+}
+// https://github.com/aws/amazon-cognito-identity-js/blob/master/src/AuthenticationHelper.js#L22
+const char* initN = 
+    "FFFFFFFFFFFFFFFFC90FDAA22168C234C4C6628B80DC1CD1"
+    "29024E088A67CC74020BBEA63B139B22514A08798E3404DD"
+    "EF9519B3CD3A431B302B0A6DF25F14374FE1356D6D51C245"
+    "E485B576625E7EC6F44C42E9A637ED6B0BFF5CB6F406B7ED"
+    "EE386BFB5A899FA5AE9F24117C4B1FE649286651ECE45B3D"
+    "C2007CB8A163BF0598DA48361C55D39A69163FA8FD24CF5F"
+    "83655D23DCA3AD961C62F356208552BB9ED529077096966D"
+    "670C354E4ABC9804F1746C08CA18217C32905E462E36CE3B"
+    "E39E772C180E86039B2783A2EC07A28FB5C55DF06F4C52C9"
+    "DE2BCBF6955817183995497CEA956AE515D2261898FA0510"
+    "15728E5A8AAAC42DAD33170D04507A33A85521ABDF1CBA64"
+    "ECFB850458DBEF0A8AEA71575D060C7DB3970F85A6E1E4C7"
+    "ABF5AE8CDB0933D71E8C94E04A25619DCEE3D2261AD2EE6B"
+    "F12FFA06D98A0864D87602733EC86A64521F2B18177B200C"
+    "BBE117577A615D6C770988C0BAD946E208E24FA074E5AB31"
+    "43DB5BFCE0FD108E4B82D120A93AD2CAFFFFFFFFFFFFFFFF";
+// https://github.com/aws/amazon-cognito-identity-js/blob/master/src/AuthenticationHelper.js#
+const char* generator = "2";
+
+
+class SRPClient {
+    public:
+    SRPClient(const std::string& N_hex, const std::string& g):N(BN_new()), g(BN_new()), a(nullptr), A(BN_new()), ctx(BN_CTX_new())
+    {
+        //convert the hex string N_hex into a BIGNUM (OpenSSL's representation of large integers)
+        AWS_FATAL_ASSERT(BN_hex2bn(&N, N_hex.c_str()));
+        //convert the decimal string g into a BIGNUM
+        AWS_FATAL_ASSERT(BN_dec2bn(&this->g, g.c_str()));
+    
+        generatePrivateKey(256);
+    }
+    ~SRPClient()
+    {
+        BN_free(N);
```

Comment:
yeah we cant be using linux/openssl symbols otherwise this wont work on windows. also we dont want to make the tests depedent on being run in linux or a direct open ssl depedency. 

[For crypto try to only use our utility classes](https://github.com/aws/aws-sdk-cpp/tree/main/src/aws-cpp-sdk-core/include/aws/core/utils/crypto), which invokes [aws-c-cal](https://github.com/awslabs/aws-c-cal) instead of taking on a direct crypto dependency.

any crypto related functionality should be flowing through the CRT